### PR TITLE
major refactor

### LINKED ISSUE #4: Refactor all code to use ESM modules
## Scope

- The current code still uses CJS; but we should use ESM module syntax

## Deliverables

Adjust all code to use ESM

## Time estimate

- 1h or less

### LINKED ISSUE #6: Correct async await logic in code base
## Scope

- E.g. I believe that in [this function](https://github.com/rugpullindex/metrics/blob/4642dfd4864c9be6939f63c8b7a3769e5945c778/index.js#L10), the JS async await logic is flawed as probably `client.guilds.cache.get(guildId).memberCount` should be awaited before returning
- Let's not mix events and promises: https://github.com/rugpullindex/metrics/blob/4642dfd4864c9be6939f63c8b7a3769e5945c778/index.js#L19
- don't use extra functions in tests: https://github.com/rugpullindex/metrics/blob/4642dfd4864c9be6939f63c8b7a3769e5945c778/test/index_test.js#L13

## Deliverables

- Generally clean up asynchronous logic in whole repo such that it makes sense.
- Additionally, consider using `cross-fetch` over `discord.js`. It seems we won't need the huge discord.js library for a simple HTTP request

## time estimate

- Anywhere between 2-3h

### PR DESCRIPTION
**Pull request by il3ven**
Fixes #6, Fixes #4 

Changes:
- Using REST API instead of discord.js
- Rewrite tests that mock fetch using esmock
- Add a `package.json`, `.gitignore`
- Removed `CLIENT_ID` from `.env`
- Added `BASE_URL` in `.env`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "nodeArguments": [
+      "--loader=esmock"
+    ]
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/rugpullindex/metrics.git"
+  },
+  "author": "Rug Pull Index",
+  "license": "ISC",
```
Comment (TimDaub) [inline package.json:20]: GPL-3.0-only

File: package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  "bugs": {
+    "url": "https://github.com/rugpullindex/metrics/issues"
+  },
+  "homepage": "https://github.com/rugpullindex/metrics#readme",
+  "dependencies": {
+    "cross-fetch": "^3.1.4",
```
Comment (TimDaub) [inline package.json:26]: I prefer to strictly pin my requirements to e.g. circumvent supply chain attacks from npm. I also [documented](https://github.com/TimDaub/commands/blob/93b1a963a94a5be100b73181f7ae7502fb9dc5d6/npm-config#L18) how you can make this a default in npm. OK?

**Review by TimDaub — CHANGES_REQUESTED** [review]
Looks really good! I just commented on two minor issues.

File: package.json (hunk 2 above)
Comment (il3ven) [reply package.json:26]: Okay. Do we need to change only dependencies or devDependencies too?

Comment (TimDaub) [reply package.json:26]: IMO dependencies are good enough.

Comment (il3ven) [reply package.json]: Done the requested changes. I also realized that `.env` is now in `.gitignore`. Thus renamed it `.env.example` so we can track the changes in it.

**Review by TimDaub — APPROVED** [review]

**Pull request merged**